src/main/java/org/broadinstitute/hellbender/tools/spark/bwa/BwaSpark.java

```diff
@@ -40,15 +47,25 @@ public boolean requiresReads() {
     }
 
     @Override
-    protected void runTool(final JavaSparkContext ctx) {
+    public List<ReadFilter> getDefaultReadFilters() {
+        // 1) unmapped or neither secondary nor supplementary and 2) has some sequence
+        return Collections.singletonList(ReadFilterLibrary.PRIMARY_ALIGNMENT.and(ReadFilterLibrary.SEQ_IS_STORED));
```

Comment:
Just imitating what I saw in WellformedReadFilter.  Is that one immune to inspection?  NB, if I don't allow unmapped reads though the PrimaryLineReadFilter I'll definitely need an OR node.  So if we can't read through AND and OR nodes, we'll still be stuck.

I'm not sure I buy all this command line control of filters, anyway.  The filters specified for the tool are the minimal set, and if you go futzing with them, you'll likely cause an unpleasant exception.  I suspect that this might actually be the case for quite a few tools.  But that's just an aside.

I disagree about adding HAS_MATCHING_BASES_AND_QUALS:  It's not necessary for this tool, and the set of downstream tools that might or might not be run is a mystery.  This tool is a stand-alone from my point of view.  But I defer to your judgement if you'd like to insist.